### PR TITLE
HPCC-29008 Add logging to Dali filesDeprecatedUser injection

### PR DESCRIPTION
**Pull request by RussWhitehead**
Log the object whose permissions are being requested, in order to help determine the component sending the Dali request without credentials

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [x] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

File: dali/server/daldap.cpp [hunk 1]
```diff
@@ -139,7 +139,7 @@ class CDaliLdapConnection: implements IDaliLdapConnection, public CInterface
         {
             username.append(filesdefaultuser);
             decrypt(password, filesdefaultpassword);
-            OWARNLOG("Missing credentials, injecting deprecated filesdefaultuser");
+            OWARNLOG("Missing credentials, injecting deprecated filesdefaultuser for request %s %s", key ? key : "NULL", obj ? obj : "NULL");
```
Comment (jakesmith) [inline dali/server/daldap.cpp:142]: minor: jstring's nullText is handy/preferable and standarizes the output for situations like this.

**Review by jakesmith — COMMENTED** [review]
@RussWhitehead - 1 minor, but could be merged as is.

Not sure re. targeting 8.8 though, @ghalliday ?

**Comment by RussWhitehead** [comment]
@jakesmith Didn't know about nullText, but now I do.
I targeted 8.8.x because Lisa Frates is having trouble with 8.6.x and I want to make this available to her ASAP.
Let me know if there is a more  appropriate release that you prefer

**Comment by jakesmith** [comment]
> @jakesmith Didn't know about nullText, but now I do. I targeted 8.8.x because Lisa Frates is having trouble with 8.6.x and I want to make this available to her ASAP. Let me know if there is a more appropriate release that you prefer

Let's leave it to @ghalliday to decice, but I think they should not go to 8.8.

**Review by jakesmith — APPROVED** [review]

**Review by ghalliday — APPROVED** [review]
Will merge into 8.8, but probably will not create a build just for this change.

**Pull request merged**